torch/csrc/jit/codegen/cuda/test/test_gpu.cpp

```diff
@@ -25555,6 +25555,77 @@ TEST_F(NVFuserTest, FusionPredicateUnshare_CUDA) {
   testValidate(fusion, {out}, {t0}, {t0}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, FusionMergeBroadcastingTrivialReduction1_CUDA) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  TensorView* tv0 = makeConcreteTensor({1, 1});
+  TensorView* tv1 = makeConcreteTensor({-1});
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  auto tv2 = sum(tv0, {1});
+  auto tv3 = add(tv2, tv1);
+  fusion->addOutput(tv3);
+
+  tv2->merge(0);
```

Comment:
I think the test should be merging the dim mapped to trivial reduction in a producer and propagating towards consumer. That's the issue we typically face in scheduling, as if the dim isn't in the consumer and we propagate towards the trivial reduction from the consumer direction it just gets removed.

Reply:
fixed